language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -386,9 +375,40 @@ impl LibraVM {
             )
             .map_err(|e| e.into_vm_status())?;
 
+        let change_set = match txn.payload() {
```

Comment:
I wonder if we could work by "de-structuring" the transaction rather than working by ref and making a copy of something we are about to throw away. 
But maybe the API sadly does not allow that?